lib/modules/datasource/metadata.ts

```diff
@@ -114,12 +125,15 @@ export function addMetaData(
   ) {
     dep.sourceUrl = dep.changelogUrl;
   }
-  // prettier-ignore
-  if (dep.homepage?.includes('github.com')) { // lgtm [js/incomplete-url-substring-sanitization]
-    if (!dep.sourceUrl) {
-      dep.sourceUrl = dep.homepage;
+
+  if (dep.homepage) {
+    const parsedHomePage = parseUrl(dep.homepage);
+    if (parsedHomePage?.hostname.includes('github')) {
```

Comment:
we should also accept gitlab

Reply:
does this count as a behavior change that should be handled in the main PR https://github.com/renovatebot/renovate/pull/16166 ?

Reply:
sure, let's do it separately